### DISCUSSION TITLE
web-ext browser.tabs.executeScript not supported?

### DISCUSSION THREAD

**Discussion by avalanche1** [parcel-bundler/parcel#5887] [Q&A] [answered]
Correct Me if I'm wrong, but it looks like `browser.tabs.executeScript` [api](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/tabs/executeScript) is not supported by the bundler.
Neither
```ts
      void browser.tabs.executeScript({
        file: "../../../../entry/content-script/cs-add-badges.ts", // source format
      });
```
nor
```ts
      void browser.tabs.executeScript({
        file: "cs-add-badges.js", // dist format
      });

```
works.

Because Parcel ignores this file reference - later on runtime I get an error in extn background script -when it tries to inject- that this file is missing.

I would expect the bundler to analyze file reference in this api and either notify on build, or bundle the file.
Is this not supported?

For now I have to manually build and put the file into the `dist` folder.

**Comment by mischnic** [comment, marked as answer]
-> https://github.com/parcel-bundler/parcel/issues/5758
